extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/InterfaceConfigPropertiesUtil.java

```diff
@@ -0,0 +1,237 @@
+                                + " is not a getter method since it returns void");
+                    }
+
+                    /*
+                     * TODO examine if returning objects can be supported - currently it isn't supported because we don't try to
+                     * create the return object
+                     * and instead just rely on what MP Config gives us back
+                     */
+
+                    NameAndDefaultValue nameAndDefaultValue = determinePropertyNameAndDefaultValue(method);
+                    String fullConfigName = prefixStr + "." + nameAndDefaultValue.getName();
+                    try (MethodCreator methodCreator = interfaceImplClassCreator.getMethodCreator(method.name(),
+                            method.returnType().name().toString())) {
+
+                        ResultHandle config = methodCreator.readInstanceField(configField, methodCreator.getThis());
+                        String defaultValueStr = nameAndDefaultValue.getDefaultValue();
+                        if (DotNames.OPTIONAL.equals(returnType.name())) {
+                            if (defaultValueStr != null) {
+                                /*
+                                 * it doesn't make to use @ConfigProperty(defaultValue="whatever") on a method that returns
+                                 * Optional
+                                 * since the result in this case isn't "optional", but there is always a value
+                                 */
+                                throw new IllegalArgumentException(
+                                        "Annotating a method returning Optional with @ConfigProperty and setting defaultValue is not supported. Offending method is "
+                                                + method.name() + " of interface" + ifaceDotName);
+                            }
+
+                            // use config.getOptionalValue to obtain the result
+
+                            Type genericType = determineSingleGenericType(returnType,
+                                    method.declaringClass().name());
+                            ResultHandle result = methodCreator.invokeInterfaceMethod(
+                                    MethodDescriptor.ofMethod(Config.class, "getOptionalValue", Optional.class, String.class,
+                                            Class.class),
+                                    config, methodCreator.load(fullConfigName),
+                                    methodCreator.loadClass(genericType.name().toString()));
+                            methodCreator.returnValue(result);
+                        } else {
+                            if (defaultValueStr != null) {
+                                /*
+                                 * The effect this will have is to add a Config Source with a lower priority thus ensuring
+                                 * that when pull a value of using config.getValue(fullConfigName) the default value
```

Comment:
This sentence is weird but I'm not sure what needs fixing.

Reply:
Hopefully it's better now